lib/zypper.pm

```diff
@@ -44,4 +50,106 @@ sub wait_quit_zypper {
     assert_script_run('until ! pgrep \'zypper|purge-kernels|rpm\' > /dev/null; do sleep 10; done', 600);
 }
 
+
+=head2 zypper_call
+
+ zypper_call($command [, exitcode => $exitcode] [, timeout => $timeout] [, log => $log] [, dumb_term => $dumb_term]);
+
+Function wrapping 'zypper -n' with allowed return code, timeout and logging facility.
+First parammeter is required command, all others are named and provided as hash
+for example:
+
+ zypper_call("up", exitcode => [0,102,103], log => "zypper.log");
+
+ # up        --> zypper -n up --> update system
+ # exitcode  --> allowed return code values
+ # log       --> capture log and store it in zypper.log
+ # dumb_term --> pipes through cat if set to 1 and log is not set. This is a  workaround
+ #               to get output without any ANSI characters in zypper before 1.14.1. See boo#1055315.
+
+C<dumb_term> will default to C<is_serial_terminal()>.
+=cut
+sub zypper_call {
+    my $command          = shift;
+    my %args             = @_;
+    my $allow_exit_codes = $args{exitcode} || [0];
+    my $timeout          = $args{timeout}  || 700;
+    my $log              = $args{log};
+    my $dumb_term        = $args{dumb_term} // is_serial_terminal;
+
+    my $printer = $log ? "| tee /tmp/$log" : $dumb_term ? '| cat' : '';
+    die 'Exit code is from PIPESTATUS[0], not grep' if $command =~ /^((?!`).)*\| ?grep/;
+
+    # Michael zypper poc
+    my $current_terminal = current_console();
+    if ($current_terminal eq "root-console") {
+        # is root, move to serial
+        opensusebasetest::select_serial_terminal();
+    }
+    my $new_terminal = current_console();
+    if ($current_terminal eq $new_terminal) {
+        # is root, move to serial
+        die "Nothing changed, still: $new_terminal";
```

Comment:
This might fail on systems that don't have serial console (I think svirt-xen is one of those)